packages/editor-ui/src/mixins/workflowHelpers.ts

```diff
@@ -1094,7 +1094,7 @@ export const workflowHelpers = defineComponent({
 				const templateId = this.$route.query.templateId;
 				if (templateId) {
 					this.$telemetry.track('User saved new workflow from template', {
-						template_id: templateId,
+						template_id: isNaN(+templateId) ? templateId : +templateId,
```

Comment:
This is personal opinion, but I feel the `+` operator for converting something to a number is such an obscure pattern, as it's not very obvious that there's a conversion happening here. Could we perhaps use `parseInt` or `Number` instead? They do [behave a bit differently](https://batfrontend.com/javascript/parseint-vs-number-vs/), but it shouldn't matter in this case.

```suggestion
						const templateIdNum = parseInt(templateId, 10)
						template_id: isNaN(templateIdNum) ? templateId : templateIdNum,
```

Reply:
Thanks @tomi! This makes sense but decided to go with the unary operator here since, when going over this with @nik8n, we decided that we don't want invalid ids such as `123abc` to be cast to `123`. Using `+` these will also be flagged as `NaN`.
Also, there is a transformation on the rudderstack side that does this check and converts ids to numbers

Reply:
Gotcha, makes sense 👍